[DOC] Binary versions info (#1266)
oneapi-src/oneDAL: docs/source/build_app/build-application.rst

            - onedal_core.lib, onedal_sequential.lib
            - onedal_core.lib, onedal_thread.lib
          * - Dynamic linking
            - onedal_core_dll.1.lib

>             - onedal_core_dll.1.lib

Reviewer: i think this is typo that should be corrected - @Alexsandruss  could you please fix this?

>             - onedal_core_dll.1.lib

Author: I don't think that it's typo actually, merging of these cells are better, but blocked since list-table don't support it.

>    Number in dynamic library file name is major binary version of library. See `Dynamic library versions`_ to find that major version means.

Author: Corrected

            - libonedal_core.so, libonedal_sequential.so
            - libonedal_core.so, libonedal_thread.so

>      There are three available file names for dynamic linking: with major and minor binary versions, with major version or without versions.

Reviewer: worth adding mention of semantic versioning https://semver.org/

And also recommendation to not use unversioned library for production use

Author: Mention and recommendation were added

>      There are three available file names for dynamic linking: with major and minor binary versions, with major version or without versions.

Author: Corrected

>      They are implemented through chain of Linux soft links:

Author: Corrected

     libonedal_*.so (soft link) -> libonedal_*.so.{major version} (soft link) -> libonedal_*.so.{major version}.{minor version} (actual library file)

>      See `Dynamic library versions`_ to find that major and minor versions mean.

Author: Corrected

     clang++ -fsycl -DONEAPI_DAAL_USE_MKL_GPU_GEMM my_first_daal_program.cpp -Wl,
     --start-group <install dir>/daal/latest/lib/intel64/libonedal_core.a <install dir>/daal/latest/lib/intel64/libonedal_sequential.a -lpthread -ldl -lOpenCL -foffload-static-lib=<install dir>/daal/latest/lib/intel64/libonedal_sycl.a -Wl,--end-group

Dynamic library versions

> ---------------------

Reviewer: we also provide this versions as part of env scripts - should be worth mentioning.

Author: Mention was added

> ---------------------

Reviewer 2: this should be the same length as the title above it

Author: Corrected

> Some dynamic library files and links have major and minor binary versions in names.

Author: Corrected

When new library version without ABI compatibility to previous versions are going to be released, major version is incremented and minor is set to 0.

> When new library version with ABI compatibility to previous versions and additional API are going to be released, minor version is incremented.

Reviewer 3: Incorrect statements. In fact the library follows semantic versioning instead.

Author: Rewritten to align with semantic versioning

     clang++ -fsycl -DONEAPI_DAAL_USE_MKL_GPU_GEMM my_first_daal_program.cpp -Wl,
     --start-group <install dir>/daal/latest/lib/intel64/libonedal_core.a <install dir>/daal/latest/lib/intel64/libonedal_sequential.a -lpthread -ldl -lOpenCL -foffload-static-lib=<install dir>/daal/latest/lib/intel64/libonedal_sycl.a -Wl,--end-group

Dynamic library versions
------------------------

The library follows `semantic versioning <https://semver.org>`_, so some dynamic library files and links have major and minor binary versions in their names.

>  - Minor version is incremented if new functionality has been introduced to the public API and backwards compatibility has been saved.

Reviewer 2: ```suggestion
 - The minor version is incremented if new functionality is introduced to the public API and backwards compatibility is preserved.
```

Author: Corrected

>  - Major version is incremented and minor is set to 0 if any backwards incompatible changes has been introduced.

Reviewer 2: ```suggestion
 - The major version is incremented and the minor version is set to 0 if any backwards incompatible changes are introduced.
```

Author: Corrected

> Usage of dynamic library without any versions in production is not recommended.

Reviewer 2: ```suggestion
It is not recommended to use dynamic library files without any versioning in production.
```

Author: Corrected

> Environment scripts are setting binary versions in DAL_MAJOR_BINARY and DAL_MINOR_BINARY variables on call.

Reviewer 2: ```suggestion
Environment scripts set binary versions in ``DAL_MAJOR_BINARY`` and ``DAL_MINOR_BINARY`` variables on call.
```

Author: Corrected
